lib/swoosh/mailer.ex

```diff
@@ -106,6 +107,16 @@ defmodule Swoosh.Mailer do
       defp parse_config(config) do
         Mailer.parse_config(@otp_app, __MODULE__, @mailer_config, config)
       end
+
+      defp instrument(response={tag, term}, key, metadata \\ %{}) do
+        status = if(tag == :ok, do: :success, else: :failure)
+        event = [:swoosh, key, status]
+        metadata = Map.merge(metadata, %{mailer: __MODULE__})
+
+        :telemetry.execute(event, %{}, metadata)
```

Comment:
I think would be better if we use [`:telemetry.span/3`](https://hexdocs.pm/telemetry/telemetry.html#span/3). It will emit the events in a more standardized way, following what others libraries do like oban, redix, finch etc. 

And I think we would probably benefit of publishing all the available information in the event, as it allows the listener to use wherever information makes sense to the application and monitoring tool been used.

```elixir
metadata = %{email: email, config: config, mailer: __MODULE__}

:telemetry.span([:swoosh, :deliver], metadata, fn ->
  case Mailer.deliver(email, parse_config(config)) do
    {:ok, result} ->
      {{:ok, response}, Map.put(metadata, :result, result)}

    {:error, reason} ->
      {{:error, reason}, Map.put(metadata, :error, reason)}
  end
end)
```

Reply:
Thanks @thiamsantos, I've updated the PR.